src/drive/web/modules/move/MoveModal.jsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Modal, Button, withBreakpoints } from 'cozy-ui/react'
+import { Query } from 'cozy-client'
+import Topbar from 'drive/web/modules/layout/Topbar'
+import { ROOT_DIR_ID, TRASH_DIR_ID } from 'drive/constants/config'
+import Alerter from 'cozy-ui/react/Alerter'
+
+import cx from 'classnames'
+
+import FileList from 'drive/web/modules/filelist/FileList'
+import FileListHeader, {
+  MobileFileListHeader
+} from 'drive/web/modules/filelist/FileListHeader'
+import FileListBody from 'drive/web/modules/filelist/FileListBody'
+import FileListRows from 'drive/web/modules/filelist/FileListRows'
+import File from 'drive/web/modules/filelist/File'
+import fileListStyles from 'drive/styles/filelist'
+
+import {
+  Breadcrumb,
+  PreviousButton,
+  renamePathNames
+} from 'drive/web/modules/navigation/Breadcrumb'
+import getFolderPath from 'drive/web/modules/navigation/getFolderPath'
+
+const MoveTopbar = withBreakpoints()(
+  ({ navigateTo, path, breakpoints: { isMobile } }) => (
+    <Topbar hideOnMobile={false}>
+      {path.length > 1 &&
+        isMobile && (
+          <PreviousButton onClick={() => navigateTo(path[path.length - 2])} />
+        )}
+      <Breadcrumb path={path} onBreadcrumbClick={navigateTo} opening={false} />
+      <div role="toolbar">add folder</div>
+    </Topbar>
+  )
+)
+
+class MoveModal extends React.Component {
+  state = {
+    folderId: ROOT_DIR_ID
+  }
+
+  navigateTo = folder => {
+    this.setState({ folderId: folder.id })
+  }
+
+  moveEntries = async () => {
+    const { entries, onClose } = this.props
+    const { client } = this.context
+    const { folderId } = this.state
+
+    const entry = entries[0]
+
+    try {
+      await client
+        .collection('io.cozy.files')
+        .updateFileMetadata(entry._id, { dir_id: folderId })
+      Alerter.info('*Thing* has been moved to *destination*', {
+        buttonText: 'cancel',
+        buttonAction: () => console.log('cancel move plz')
+      })
+    } catch (e) {
+      console.warn(e)
+      Alerter.error('move error')
+    } finally {
+      onClose()
+    }
+  }
+
+  sortData = data => {
+    const folders = data.filter(
+      entry => entry.type === 'directory' && entry.id !== TRASH_DIR_ID
+    )
+    const files = data.filter(entry => entry.type !== 'directory')
+
+    return folders.concat(files)
+  }
+
+  buildBreadcrumbPath = data =>
+    renamePathNames(
+      getFolderPath({
+        ...data,
+        parent: data.relationships.parent.data
+      }),
+      '',
+      this.context.t
+    )
+
+  render() {
+    const { onClose } = this.props
+    const { client, t } = this.context
+    const { folderId } = this.state
+
+    const contentQuery = client =>
+      client
+        .find('io.cozy.files')
+        .where({ dir_id: folderId })
+        .sortBy({ name: 'asc' })
+
+    const breadcrumbQuery = client => client.get('io.cozy.files', folderId)
```

Comment:
Can we create the request constant somewhere else ? We don't need to recreate these request everytime we render the component, right ?

Reply:
Never mind, didn't see the arrow function :(

Reply:
Definitely something I want to look into when cleaning! But they need to be recreated when `folderId` changes, which may or may not be the only time the component re-renders.